mypy/checkexpr.py

```diff
@@ -327,7 +327,20 @@ def infer_arg_types_in_context2(
         for i, actuals in enumerate(formal_to_actual):
             for ai in actuals:
                 if arg_kinds[ai] not in (nodes.ARG_STAR, nodes.ARG_STAR2):
-                    res[ai] = self.accept(args[ai], callee.arg_types[i])
+                    callee_arg_type = callee.arg_types[i]
+                    arg_type = self.accept(args[ai], callee_arg_type)
+                    if (isinstance(arg_type, CallableType) and
```

Comment:
This doesn't seem to fix this for more general argument types. For example, what if the formal argument type is `List[Callable[...]]`? We could look into the types recursively, but not sure if that's the right thing to do (and it might be inefficient). I thought about this for a bit and this is far from clear to me.